src/consensus/encode.rs

```diff
@@ -553,21 +583,25 @@ impl<D: Decoder, T: Decodable<D>> Decodable<D> for Vec<T> {
     }
 }
 
-impl<S: Encoder, T: Encodable<S>> Encodable<S> for Box<[T]> {
```

Comment:
So this is effectively a breaking change that doesn't allow encoding (boxed) slices of encodables anymore? Are you sure this doesn't affect certain users?

Reply:
This is the least interesting of all the generic impls that are being dropped :)

Reply:
The argument that it's meant for consensus is quite a convincing one. We shouldn't support any types more than the ones ever used by consensus. And especially having very broad generic types (like `[T]`) prevents some users to implement such constructions using their own types.